pymc3/distributions/multivariate.py

```diff
@@ -255,56 +262,53 @@ def random(self, point=None, size=None):
         -------
         array
         """
-        if size is None:
-            size = tuple()
+        size = to_tuple(size)
+
+        param_attribute = "chol_cov" if self._cov_type == "chol" else self._cov_type
+        mu, param = draw_values([self.mu, getattr(self, param_attribute)], point=point, size=size)
+        if tuple(self.shape):
+            dist_shape = tuple(self.shape)
         else:
-            if not isinstance(size, tuple):
-                try:
-                    size = tuple(size)
-                except TypeError:
-                    size = (size,)
+            if is_fast_drawable(self.mu) or point:
+                batch_shape = mu.shape[:-1]
+            else:
+                batch_shape = mu.shape[len(size) : -1]
+            dist_shape = batch_shape + param.shape[-1:]
```

Comment:
You are forgetting to consider `self.shape` in these conditions.

Reply:
This is what I am doing actually - 

1. If `self.shape` is given, (which is mostly the case), the distribution shape is assumed to be `self.shape`.

2. If `self.shape` is not given, (probably using `pm.MvNormal.dist`), the distribution shape is inferred from batches in `mu` and event in `cov`. So, that's why I am using `self.shape` as pointed out here.

After drawing the samples, `mu` is padded with extra dimensions on the left to match the output shape ( i.e. sample + batch + event).